src/rotatez.jl

```diff
@@ -0,0 +1,508 @@
+        linearinterp!(interp, workvec)
+        mul!((@view output[:, i]), interp', (@view img[:, i])) # need mul! to avoid allocating
+    end
+    return output
+end
+
+
+"""
+    rotate_y!(output, img, sin_θ, workvec, interp)
+"""
+function rotate_y!(
+    output::AbstractMatrix{<:RealU},
+    img::AbstractMatrix{<:RealU},
+    sin_θ::RealU,
+    workvec::AbstractVector{<:RealU},
+    interp::SparseInterpolator,
+)
+
+    len = length(workvec)
+    c_x = (len + 1)/2 # center of xi
+    idx = 1:len
+
+    for i in idx
+        workvec .= idx[i]
+        broadcast!(-, workvec, workvec, c_x)
+        broadcast!(*, workvec, workvec, sin_θ)
+        broadcast!(+, workvec, workvec, idx)
+        linearinterp!(interp, workvec)
+        mul!((@view output[i, :]), interp, (@view img[i, :]))
+    end
+    return output
+
+end
+
+
+"""
+    rotate_y_adj!(output, img, sin_θ, workvec, interp)
+"""
+function rotate_y_adj!(
+    output::AbstractMatrix{<:RealU},
+    img::AbstractMatrix{<:RealU},
+    sin_θ::RealU,
+    workvec::AbstractVector{<:RealU},
+    interp::SparseInterpolator,
+)
+
+    len = length(workvec)
+    c_x = (len + 1)/2 # center of xi
+    idx = 1:len
+
+    for i in idx
+        workvec .= idx[i]
+        broadcast!(-, workvec, workvec, c_x)
+        broadcast!(*, workvec, workvec, sin_θ)
+        broadcast!(+, workvec, workvec, idx)
+        linearinterp!(interp, workvec)
+        mul!((@view output[i, :]), interp', (@view img[i, :]))
+    end
+    return output
+
+end
+
+
+
+"""
+    rotl90!(B::AbstractMatrix, A::AbstractMatrix)
+In place version of `rotl90`, returning rotation of `A` in `B`.
+"""
+function rotl90!(B::AbstractMatrix, A::AbstractMatrix)
+    ind1, ind2 = axes(A)
+    n = first(ind2) + last(ind2)
+    for i = axes(A, 1), j = ind2
+        B[n - j, i] = A[i, j]
+    end
+    return B
+end
+
+
+"""
+    rotr90!(B::AbstractMatrix, A::AbstractMatrix)
+In place version of `rotr90`, returning rotation of `A` in `B`.
+"""
+function rotr90!(B::AbstractMatrix, A::AbstractMatrix)
+    ind1, ind2 = axes(A)
+    m = first(ind1) + last(ind1)
+    for i = ind1, j = axes(A, 2)
+        B[j, m - i] = A[i, j]
+    end
+    return B
+end
+
+
+"""
+    rot180!(B::AbstractMatrix, A::AbstractMatrix)
+In place version of `rot180`, returning rotation of `A` in `B`.
+"""
+function rot180!(B::AbstractMatrix, A::AbstractMatrix)
+    ind1, ind2 = axes(A,1), axes(A,2)
+    m, n = first(ind1)+last(ind1), first(ind2)+last(ind2)
+    for j=ind2, i=ind1
+        B[m-i,n-j] = A[i, j]
+    end
+    return B
+end
+
+
+"""
+    rot_f90!(output, img, m)
+In-place version of rotating an image by 90/180/270 degrees
+"""
+function rot_f90!(
+    output::AbstractMatrix,
+    img::AbstractMatrix,
+    m::Int,
+)
+    if m == 0
+        output .= img
+    elseif m == 1
+        rotl90!(output, img)
+    elseif m == 2
+        rot180!(output, img)
+    elseif m == 3
+        rotr90!(output, img)
+    else
+        throw("invalid m!")
```

Comment:
please add a `@test_throws` case like i did for `rot_f90!` in test/rotate3.jl
here and below.